packages/debug/src/browser/debug-session-manager.ts

```diff
@@ -166,6 +166,7 @@ export class DebugSessionManager {
     async start(options: DebugSessionOptions): Promise<DebugSession | undefined> {
         return this.progressService.withProgress('Start...', 'debug', async () => {
             try {
+                await this.editorManager.saveAll();
```

Comment:
you should use `ApplicationShell`, not all editors (widgets implementing Saveable) are code editors